escrow/contracts/Escrow.cdc

```diff
@@ -196,23 +188,23 @@ pub contract Escrow {
         }
 
         // Call addEntry.
-        pub fun addEntryToLeaderboard(nft: @NonFungibleToken.NFT, leaderboardName: String, depositCap: Capability<&{NonFungibleToken.CollectionPublic}>) {
+        pub fun addEntryToLeaderboard(nft: @NonFungibleToken.NFT, leaderboardName: String, ownerAddress: Address, metadata: {String: AnyStruct}) {
             let leaderboard = &self.leaderboards[leaderboardName] as &Leaderboard?
             if leaderboard == nil {
                 panic("Leaderboard does not exist with this name")
             }
 
-            leaderboard!.addEntryToLeaderboard(nft: <-nft, leaderboardName: leaderboardName, depositCap: depositCap)
+            leaderboard!.addEntryToLeaderboard(nft: <-nft, ownerAddress: ownerAddress, metadata: metadata)
         }
 
         // Calls transferNftToCollection.
-        pub fun transferNftToCollection(leaderboardName: String, nftID: UInt64) {
+        pub fun transferNftToCollection(leaderboardName: String, nftID: UInt64, depositCap: Capability<&{NonFungibleToken.CollectionPublic}>) {
```

Comment:
Curious about reason for having capability passed in here vs it being on the entry as it was previously?
I suppose the functional difference is in this pattern it's up to the _admin_ to know where NFT should be returned to, where as previously the user could have specified alternate collection paths when the escrowed the moment.

Reply:
Correct. Rather than taking the depositCap from the owner we know allow the admin to borrow the public capability from the user on transfer and check if the moment belongs to them before transfer.

Reply:
In the original approach, Josh pointed out that the capability could be invalidated by the user making it impossible to transfer the nft back to the user